lib/runners/dart.js

```diff
@@ -0,0 +1,171 @@
+'use strict'
+
+const shovel = require('../shovel');
+const util = require('../util');
+const config = require('../config');
+const fs = require('fs');
+const path = require('path');
+const temp = require('temp').track();
+
+const dir = '/tmp/dart';
+
+module.exports.run = function run(opts, cb) {
+  shovel.start(opts, cb, {
+    solutionOnly: function(run) {
+      _createSolution(opts).then(file => {
+        // Run
+        run({
+          name: 'dart',
+          args: [file],
+          options: {
+            cwd: dir
+          }
+        });
+      });
+    },
+    testIntegration: function(run) {
+      // test is the only one supported right now...
+      if (opts.testFramework != 'test')
+        throw 'Test framework is not supported';
+
+      // Create the spec test
+      _createSpec(opts).then(file => {
+        // Run
+        run({
+          name: 'pub',
+          args: ['run', 'test', file],
+          options: {
+            cwd: dir
+          }
+        });
+      });
+    },
+    sanitizeStdOut: function(stdout) {
+      if (opts.fixture) return _format(stdout)
+      else return stdout;
+    }
+  });
+};
+
+// Create the solution file
+var _createSolution = function(opts) {
+  // Set the code, _block() will stop user adding imports
+  // that are not provided in setup
+  let code = `
+    ${opts.setup ? opts.setup : ''}
+    _block() {}
```

Comment:
Oh hmm I must have dreamt up this requirement when I was looking at the `setup` part. In that case probably that should be changed to allow it, but its just that when something is added in setup it will block that possibility because the imports must be at the top else its an error. So if an author puts some variable or any other functions etc into the setup no user will be able to add imports. 

Probably thats not a big deal. If the convention is to not do this then I should remove this and then switch the spec one up up again to put the code section back above main.